### DISCUSSION TITLE
Question concerning box shadow generator and related tools

### DISCUSSION THREAD

**Discussion by schalkneethling** [mdn/mdn-community#674] [Content] [answered]
👋 I remember when I was still at Mozilla, we had a conversation about these tools and we were not 100% on what to do going forward. I notice that they are still around, but could do with some work.

Some existing issues:

- https://github.com/mdn/css-examples/issues/96
- https://github.com/mdn/css-examples/issues/93
- https://github.com/mdn/css-examples/issues/157
- https://github.com/mdn/css-examples/issues/63
- https://github.com/mdn/css-examples/issues/60

I guess we have three options:

1. Remove from MDN Web Docs, add the code to one or more repos, and archive said repos.
2. Take some time to rewrite the tools using modern JS, CSS, HTML.
3. Rewrite the tools _and_ create some microsites (could just use GitHub pages without a fancy URL).

The second option is probably, interestingly enough, the one with the least amount of work and the most benefit. If someone, for example, me 😃, would volunteer to do the work, would you all be open to that, or has another decision already been made around these tools?

Let me know, happy to discuss further.

**Comment by caugner** [comment]
> I guess we have three options:
> 
>     1. Remove from MDN Web Docs, add the code to one or more repos, and archive said repos.
> 
>     2. Take some time to rewrite the tools using modern JS, CSS, HTML.
> 
>     3. Rewrite the tools _and_ create some microsites (could just use GitHub pages without a fancy URL).
> 
> 
> The second option is probably, interestingly enough, the one with the least amount of work and the most benefit. If someone, for example, me 😃, would volunteer to do the work, would you all be open to that, or has another decision already been made around these tools?

Hi Schalk, 👋 

Great to see you around, I just checked with the content team. We definitely want to keep these tools around and it would be great to update them, but it's not a priority for us at the moment to do it ourselves. So if you're interested in volunteering to do the work, that would be awesome and much appreciated! 🙌 

How would you go about rewriting them?

Apart from functional bugs, I believe the main issues are their design (not responsive), and the lack of support for newer syntax maybe?

**Reply by schalkneethling** [reply]
Oh hai there @caugner 👋 

> Apart from functional bugs, I believe the main issues are their design (not responsive), and the lack of support for newer syntax maybe?

That was exactly my idea. 

- Clone the repo locally
- Run the demo standalone
- Update the layout so it is responsive
- Update the JS to use modern JS that is supported widely enough so that we do not have to polyfill
- Test it within the context of the page
- Open PR for review.

If that works, do you all have a preference for which tool to start with? I guess the one that has the most bugs filed as that could be an indication of the one that is used the most. Let me know. I am looking forward to contributing again 🙌

**Reply by schalkneethling** [reply]
@caugner Does the above sound like a plan?

**Reply by caugner** [reply]
@schalkneethling Thank you, that sounds like a good plan.

To make reviewing easier, I would suggest upgrading one tool at a time, and opening separate PRs for the responsive layout and the modern JS, unless these depend on each other? You can ping me this and next week, and @LeoMcA while I'm off the week after.

Here are the tools in order of popularity:

1. Color picker
2. Box-shadow generator
3. Border-image generator
4. Border-radius generator

**Reply by schalkneethling** [reply]
Great stuff!

> opening separate PRs for the responsive layout and the modern JS, unless these depend on each other?

Makes sense to me.

**Reply by schalkneethling** [reply]
I will create the relevant issues. 👍

**Reply by schalkneethling** [reply, marked as answer]
The relevant issues have been created:

- https://github.com/mdn/css-examples/issues/178
- https://github.com/mdn/css-examples/issues/179
- https://github.com/mdn/css-examples/issues/180
- https://github.com/mdn/css-examples/issues/181
